### PR TITLE
EvseSecurity module documentation

### PR DESCRIPTION
**Pull request by Pietfried**
## Describe your changes
This PR adds custom module documentation for the EvseSecurity module

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Review by AssemblyJohn — APPROVED** [review]
Looks good to me.

File: modules/EvseSecurity/doc.rst [hunk 1]
```diff
@@ -1,22 +1,69 @@
...
+
+Configuration for OCPP
+----------------------
+
+In OCPP and OCPP security profiles, the security level of the connection is specified as follows:
+
+* SecurityProfile 0: Unsecure transport without Basic Authentication
+* SecurityProfile 1: Unsecure transport with Basic Authentication
+* SecurityProfile 2: TLS with Basic authentication
+* SecurityProfile 3: TLS with client-side certificates
```
Comment (barsnick) [inline modules/EvseSecurity/doc.rst:34]: - either "unsecured" (past tense, as used in the OCA white paper for OCPP 1.6J and in OCPP 2.0.1) or "insecure" (adjective)
- inconsistent capitalization

File: modules/EvseSecurity/doc.rst [hunk 2]
```diff
@@ -1,22 +1,69 @@
...
+
+If security profile 3 is used a mutual TLS (mTLS) connection shall be established, so the CSMS server also verifies the client certificate. The ``csms_leaf_cert_directory`` and ``csms_leaf_key_directory`` need to be configured for this. These parameters specify the directory of the client certificate and key for the mTLS connection.
+
+New CSMS client certificates can be installed using the ``update_leaf_certificate`` command with the CSMS domain specified. In OCPP new client certificates are installed by using the ``CertificateSigned.req`` message.
+
+Configuration for ISO15118
+--------------------------
+
+For ISO15118 communication, the charging station provides a server endpoint to which the electric vehicle connects. The communication may be secured using TLS. TLS is mandatory for Plug&Charge. If TLS is used, the correct configuration of the ``secc_leaf_cert_directory`` and ``secc_leaf_key_directory`` is required. These directories are used to locate the server certificate and key for the ISO15118 TLS server provided by the charging station.
```
Comment (barsnick) [inline modules/EvseSecurity/doc.rst:51]: Fun fact: In ISO 15118-20, PnC expands to "Park'n'Charge" 😉

**Comment by krealyt** [comment]
Should we delete the docs directory? There is another index.rst in there that seems not to be needed.

I guess the current docs concept is: Having either the docs directory OR the docs.rst, right?

**Review by barsnick — COMMENTED** [review]
I welcome these docs!

I miss an instruction on the expected bundles and how to construct them. Trying to use certificates from CharIN (testival Jan 2024, created by Keysight), I reverse-engineered what is created by `cmake` and `make install` in everest-core in order to get libevse-security to accept them.

**Pull request merged**